containerfixture/containerfixture.go

```diff
@@ -130,7 +144,8 @@ func cleanup(t *testing.T, podName string) {
 	}
 }
 
-func WithServiceContainer(t *testing.T, ServiceManifest string, testFunc func(FixtureInfo)) {
+func WithServiceContainer(t *testing.T, serviceName, serviceDirPath, ServiceManifest string, testFunc func(FixtureInfo)) {
```

Comment:
Can we be a little bit smarter and avoid changing the function signature here?

Reply:
Do we really want to add another wrapper on top of that? We still need a service container, but we need to build it first now.

Reply:
I don't understand how did my comment imply we need another wrapper

Reply:
either way we'll need the path to the images, therefor changing the signature. unless  we decide to make assumptions (= the path to the dockerfile is always `../<service_name>`)

Reply:
you can figure out the path to the image from the YAML like Podman does.

Its `./image_name` iff `./image_name/Dockerfile` or `./image_name/Containerfile` exists (We don't really have to support `Containerfile` ATM though).

Reply:
That's the thing - the WD is not the root directory of the repo, but the directory the test resides in (running `go test ./...` will have `./splunk` as the WD when running the Splunk test, and `./kwok` when running the kwok test)
it might actually make it easier, meaning we can just build from `./Dockerfile`, but it won't necessarily work with multiple images, or future tests that may need multiple builds from different directory. 
however, we can get the caller path (`containerfixture`), and do what you suggested from the parent directory (the root dir)

Reply:
How did you manage to make `kube play` work with it? You didn't need to pass anyhthing of that sort whe you were using it.

Reply:
It may simplify things to set PWD to a fixed location when running kube play - you can figure out where to set it to via a similar trick to what I did for `scripts/runner` and then set it in the `Cmd.Dir` field whin invoking `podman`.

Reply:
- we never ran multiple images from kube play before, so it ran the `splunk` image from current directory, since it was root/splunk anyway. 
- I'm not sure I understand what in `scripts/runner` is relevant? I do intend to use `Cmd.Dir` on the calls to podman, but my current method of getting the root directory (best for podman fixed location IMO) is using `filepath.Dir` twice on `runtime.Caller(0)` (`<path_to_root>/containerfixture/containerfixture.go` -> `<path_to_root>`)
- [Parsing a YAML](https://zetcode.com/golang/yaml/) file isn't that straightforward and clean in GO; I don't mind implementing it, but since I'm off for the next week and this PR is a blocker for both Sri and Homaja, I offer getting the service name as an argument for now (as we don't have multiple-services yamls currently), will open a ticket and fix it once I'm back with a better solution.

Reply:
I'm not convinced this PR is a blocker. @srivickynesh and @hmariset can unblock themselves by manually pre-building the images before running the test.

Reply:
Modified the solution to parse the yaml and take out the image names.